diff --git a/linux/user/ihkmond.c b/linux/user/ihkmond.c
index a5ba85a..10c7bf3 100644
--- a/linux/user/ihkmond.c
+++ b/linux/user/ihkmond.c
@@ -194,7 +194,7 @@ static void* detect_hungup(void* _arg) {
 
  wait_for_removal:
 	do {
-		int nfd = epoll_wait(epfd, events, 1, arg->interval * 1000);
+		int nfd = epoll_wait(epfd, events, 1, arg->interval);
 		if (nfd < 0 && errno == EINTR)
 			continue;
 		CHKANDJUMP(nfd < 0, -EINVAL, "epoll_wait failed\n");
