diff --git a/kernel/freeze.c b/kernel/freeze.c
index 677fca5b..0717fcac 100644
--- a/kernel/freeze.c
+++ b/kernel/freeze.c
@@ -20,6 +20,11 @@ freeze()
 	if (monitor->status_bak & IHK_OS_MONITOR_ALLOW_THAW_REQUEST) {
 		return;
 	}
+
+	if (ihk_mc_get_processor_id() != 0) {
+		ihk_mc_delay_us(1000000);
+	}
+
 	monitor->status_bak = monitor->status
 				| IHK_OS_MONITOR_ALLOW_THAW_REQUEST;
 	monitor->status = IHK_OS_MONITOR_KERNEL_FROZEN;
diff --git a/kernel/syscall.c b/kernel/syscall.c
index e677e3f0..ecf595c3 100644
--- a/kernel/syscall.c
+++ b/kernel/syscall.c
@@ -9847,6 +9847,20 @@ SYSCALL_DECLARE(util_register_desc)
 	return 0;
 }
 
+SYSCALL_DECLARE(panic)
+{
+	panic("sys_panic: transition to panic state for IHK IF test");
+	return 0;
+}
+
+SYSCALL_DECLARE(hungup)
+{
+	kprintf("%s: enter\n",
+		__func__);
+	ihk_mc_delay_us(4000000);
+	return 0;
+}
+
 void
 reset_cputime()
 {
diff --git a/arch/arm64/kernel/include/syscall_list.h b/arch/arm64/kernel/include/syscall_list.h
index bb3f6c0..2566e6a 100644
--- a/arch/arm64/kernel/include/syscall_list.h
+++ b/arch/arm64/kernel/include/syscall_list.h
@@ -148,3 +148,5 @@ SYSCALL_DELEGATED(1069, epoll_wait)
  * EOF just after it because those are used as a
  * robust marker for the autotest patch.
  */
+SYSCALL_HANDLED(2001, panic)
+SYSCALL_HANDLED(2002, hungup)
