--- old_CMakeLists.txt	2020-09-01 18:39:46.095700972 +0700
+++ new_CMakeLists.txt	2020-09-01 18:40:05.958881896 +0700
@@ -210,6 +210,7 @@
 add_subdirectory(kernel)
 add_subdirectory(tools/mcstat)
 add_subdirectory(tools/crash)
+add_subdirectory(ihk/test/ihklib/whitebox/src/driver/mcctrl)
 
 configure_file(scripts/mcreboot-smp.sh.in mcreboot.sh @ONLY)
 configure_file(scripts/mcstop+release-smp.sh.in mcstop+release.sh @ONLY)
