--- old_CMakeLists.txt	2020-10-22 11:57:48.156786744 +0700
+++ new_CMakeLists.txt	2020-10-22 11:58:45.152187583 +0700
@@ -30,11 +30,12 @@
 endif()
 
 set(MCKERNEL_SRCS
-	init.c mem.c debug.c mikc.c listeners.c ap.c syscall.c cls.c host.c process.c
+	init.c mem.c debug.c mikc.c listeners.c ap.c syscall.c cls.c process.c
 	copy.c waitq.c futex.c timer.c plist.c fileobj.c shmobj.c zeroobj.c
 	procfs.c devobj.c sysfs.c xpmem.c profile.c freeze.c rbtree.c pager.c
 	gencore.c hugefileobj.c llist.c
 
+	${IHK_FULL_SOURCE_DIR}/test/ihklib/whitebox/src/driver/mckernel/host.c
 	${IHK_FULL_SOURCE_DIR}/ikc/manycore.c
 	${IHK_FULL_SOURCE_DIR}/ikc/master.c
 	${IHK_FULL_SOURCE_DIR}/ikc/queue.c
