--- old_CMakeLists.txt	2020-08-19 00:06:35.340610234 +0700
+++ new_CMakeLists.txt	2020-08-19 00:09:17.798522123 +0700
@@ -3,6 +3,10 @@
 	"${PROJECT_SOURCE_DIR}/linux/include/ihk/arch/${ARCH}"
 	"${PROJECT_BINARY_DIR}/linux/include"
 	"${PROJECT_BINARY_DIR}"
+        "${PROJECT_SOURCE_DIR}/test/ihklib/whitebox/include"
+	"${PROJECT_SOURCE_DIR}/test/ihklib/blackbox/include"
+        "${PROJECT_SOURCE_DIR}/test/ihklib/"
+        "${PROJECT_SOURCE_DIR}/linux/include/ihk/"
 )
 
 execute_process(COMMAND awk -F= "$1 == \"CONFIG_ARM64_64K_PAGES\" { print $2; exit; }" "${KERNEL_DIR}/.config"
@@ -18,7 +22,10 @@
 endif()
 message("PAGE_SIZE: ${PAGE_SIZE}")
 
-add_library(ihklib SHARED ihklib.c)
+#add_library(ihklib SHARED ihklib.c)
+add_library(ihklib SHARED ../../test/ihklib/whitebox/src/user/ihklib.c ../../test/ihklib/blackbox/src/cpu.c 
+                          ../../test/ihklib/blackbox/src/mem.c ../../test/ihklib/blackbox/src/os.c
+                          ../../test/ihklib/blackbox/src/linux.c)
 target_compile_definitions(ihklib PRIVATE -DPAGE_SIZE=${PAGE_SIZE})
 SET_TARGET_PROPERTIES(ihklib PROPERTIES OUTPUT_NAME ihk)
 target_link_libraries(ihklib ${LIBBFD})
