--- old_CMakeLists.txt	2020-09-08 18:13:36.287292949 +0700
+++ new_CMakeLists.txt	2020-09-08 18:13:23.415464811 +0700
@@ -20,9 +20,14 @@
 		-I${PROJECT_SOURCE_DIR}/kernel/include
 		-DMCEXEC_PATH=\\"${MCEXEC_PATH}\\"
 		${ARCH_C_FLAGS}
+                -I${CMAKE_CURRENT_SOURCE_DIR}
+                -I${PROJECT_SOURCE_DIR}/ihk/test/ihklib/whitebox/include
 	SOURCES
-		driver.c control.c ikc.c syscall.c procfs.c binfmt_mcexec.c
-		sysfs.c sysfs_files.c arch/${ARCH}/archdeps.c
+                ../../../ihk/test/ihklib/whitebox/src/driver/mcctrl/driver.c 
+                ../../../ihk/test/ihklib/whitebox/src/driver/mcctrl/control.c ../../../ihk/test/ihklib/whitebox/src/driver/mcctrl/ikc.c 
+                ../../../ihk/test/ihklib/whitebox/src/driver/mcctrl/syscall.c ../../../ihk/test/ihklib/whitebox/src/driver/mcctrl/procfs.c binfmt_mcexec.c
+                ../../../ihk/test/ihklib/whitebox/src/driver/mcctrl/sysfs.c ../../../ihk/test/ihklib/whitebox/src/driver/mcctrl/sysfs_files.c 
+                arch/${ARCH}/archdeps.c
 	EXTRA_SYMBOLS
 		${PROJECT_BINARY_DIR}/ihk/linux/core/Module.symvers
 	DEPENDS
